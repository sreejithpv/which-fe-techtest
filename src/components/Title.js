import React from 'react';
import PropTypes from 'prop-types';

const Title = ({ title  }) => (
  <h1 data-testid="title" tabIndex="0">{ title }</h1>
);

Title.propTypes = {
    title: PropTypes.string,
  };
  
  Title.defaultProps = {
    title: '',
  }

export default React.memo(Title);