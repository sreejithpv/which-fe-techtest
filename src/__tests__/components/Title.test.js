import React from 'react';
import TestRenderer from 'react-test-renderer';
import { render } from '@testing-library/react';

import Title from '../../components/Title';

describe('Title Component', () => {
  const { getByTestId } = render(<Title title="This is title component" />);

  it('displays the give title content', () => {
    const tree = getByTestId('title');
    expect(tree.innerHTML).toEqual('This is title component');
  });

  test("Render snapshot the component", () => {
    const tree = TestRenderer.create(<Title title="Snapshot title" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});