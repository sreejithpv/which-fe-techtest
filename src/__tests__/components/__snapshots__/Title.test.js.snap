// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title Component Render snapshot the component 1`] = `
<h1
  data-testid="title"
  tabIndex="0"
>
  Snapshot title
</h1>
`;
